django/contrib/gis/db/backends/base/operations.py

```diff
@@ -114,36 +114,34 @@ def get_distance(self, f, value, lookup_type):
             "Distance operations not available on this spatial backend."
         )
 
-    def get_geom_placeholder(self, f, value, compiler):
+    def get_geom_placeholder_sql(self, f, value, compiler):
         """
         Return the placeholder for the given geometry field with the given
         value. Depending on the spatial backend, the placeholder may contain a
         stored procedure call to the transformation function of the spatial
         backend.
         """
 
-        def transform_value(value, field):
+        def must_transform_value(value, field):
             return value is not None and value.srid != field.srid
 
         if hasattr(value, "as_sql"):
-            return (
-                "%s(%%s, %s)" % (self.spatial_function_name("Transform"), f.srid)
-                if transform_value(value.output_field, f)
-                else "%s"
-            )
-        if transform_value(value, f):
-            # Add Transform() to the SQL placeholder.
-            return "%s(%s(%%s,%s), %s)" % (
-                self.spatial_function_name("Transform"),
-                self.from_text,
-                value.srid,
-                f.srid,
-            )
+            sql, params = compiler.compile(value)
+            if must_transform_value(value.output_field, f):
+                transform_func = self.spatial_function_name("Transform")
```

Comment:
We can't assign to the method scope as some backends don't support SRID transforms (e.g. MySQL)